waitress/channel.py

```diff
@@ -370,6 +391,15 @@ def service(self):
                         request.close()
                     self.requests = []
                 else:
+                    # before processing a new request, ensure there is not too
+                    # much data in the outbufs waiting to be flushed
+                    # NB: currently readable() returns False while we are
+                    # flushing data so we know no new requests will come in
+                    # that we need to account for, otherwise it'd be better
+                    # to do this check at the start of the request instead of
+                    # at the end to account for consecutive service() calls
+                    if len(self.requests) > 1:
```

Comment:
Would it make sense here to kick the trigger?

Reply:
I'm not familiar with the scenarios in which kicking the trigger is required.

Reply:
It's not like the current behavior is worse than it was before. I would assume the situation is that the write returned 0 bytes written because the buffer was full, and the system would know that more data is writable before hitting a select timeout or something. If you think it's a problem then it should be kicked in `_flush_outbufs_below_high_watermark` which affects both app_iter writes as well as here between requests. I can't imagine the circumstances are different here than there.